modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaMultiVMPoolUtilCheck.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nícolas Moura
+ */
+public class UpgradeJavaMultiVMPoolUtilCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		if (content.contains(_MULTI_VM_POOL_UTIL_IMPORT)) {
+			content = StringUtil.replace(
+				content, _MULTI_VM_POOL_UTIL_IMPORT,
+				"import com.liferay.portal.kernel.cache.MultiVMPool;");
+			content = _replaceGetPortalCache(content, fileName);
+		}
+
+		return content;
+	}
+
+	private String _replaceGetPortalCache(String content, String fileName)
+		throws Exception {
+
+		content = JavaSourceUtil.addImport(
+			content,
+			"import org.osgi.service.component.annotations.Reference;");
+
+		Matcher portalCacheMatcher = _getPortalCachePattern.matcher(content);
+
+		while (portalCacheMatcher.find()) {
+			String newDeclaration = StringUtil.replace(
+				portalCacheMatcher.group(0), "MultiVMPoolUtil.getPortalCache(",
+				"(PortalCache" + portalCacheMatcher.group(1) +
+					") _multiVMPool.getPortalCache(");
+
+			content = StringUtil.replace(
+				content, portalCacheMatcher.group(0), newDeclaration);
+		}
+
+		content = StringUtil.replace(
+			content, "MultiVMPoolUtil.getPortalCache(",
+			_WARNING_CASE + " _multiVMPool.getPortalCache(");
+		content = StringUtil.replaceLast(
+			content, CharPool.CLOSE_CURLY_BRACE,
+			"\n\t@Reference\n\tprivate MultiVMPool _multiVMPool;\n}");
+
+		addMessage(
+			fileName, "Replace 'TO_BE_REPLACED' with the correct type",
+			getLineNumber(content, content.indexOf(_WARNING_CASE)));
```

Comment:
Two things here:

1. This message always gets printed in all cases. This should only be displayed when there was a warning case. Also, `getLineNumber()` would give us a weird number if there was no warning case.
2. This message only gets printed ___once___. What if there are multiple cases where the warning replacement is done?

Reply:
For the 2nd point, you have the option of just displaying a general warning message to the user without specifying the line numbers. Something like "Could not resolve types for `MultiVMPool.getPortalCache()`..." or something like that.

Reply:
Well noticed,  yeah, maybe a general warning will be better in this case. I will take a look and resent